src/tasks/AndroidAppBuilder/Templates/MonoRunner.java

```diff
@@ -146,33 +146,33 @@ public void onDestroy() {
 
     static void unzipAssets(Context context, String toPath, String zipName) {
         AssetManager assetManager = context.getAssets();
-        try {
-            InputStream inputStream = assetManager.open(zipName);
-            ZipInputStream zipInputStream = new ZipInputStream(new BufferedInputStream(inputStream));
+        try (InputStream inputStream = assetManager.open(zipName);
+            ZipInputStream zipInputStream = new ZipInputStream(new BufferedInputStream(inputStream))) {
+
             ZipEntry zipEntry;
             byte[] buffer = new byte[4096];
             while ((zipEntry = zipInputStream.getNextEntry()) != null) {
                 String fileOrDirectory = zipEntry.getName();
-                Uri.Builder builder = new Uri.Builder();
-                builder.scheme("file");
-                builder.appendPath(toPath);
-                builder.appendPath(fileOrDirectory);
-                String fullToPath = builder.build().getPath();
-                if (zipEntry.isDirectory()) {
-                    File directory = new File(fullToPath);
-                    directory.mkdirs();
-                    continue;
+                File file = new File(toPath, fileOrDirectory);
+                File parent = new File(file.getParent());
+
+                if (file.isDirectory()) {
+                    file.mkdirs();
+                }
+                else if (!parent.exists()) {
+                    parent.mkdirs();
                 }
+
+                String fullToPath = file.getAbsolutePath();
                 Log.i("DOTNET", "Extracting asset to " + fullToPath);
+
                 int count = 0;
                 FileOutputStream fileOutputStream = new FileOutputStream(fullToPath);
                 while ((count = zipInputStream.read(buffer)) != -1) {
                     fileOutputStream.write(buffer, 0, count);
                 }
-                fileOutputStream.close();
                 zipInputStream.closeEntry();
             }
-            zipInputStream.close();
```

Comment:
why did you remove closing the streams?

Reply:
This should automatically close?

```
try (InputStream inputStream = assetManager.open(zipName);
       ZipInputStream zipInputStream = new ZipInputStream(new BufferedInputStream(inputStream)))
```

Reply:
ahh, I didn't see that this changed from `try {` to `try (` :)